parquet-column/src/main/java/org/apache/parquet/schema/Type.java

```diff
@@ -363,4 +365,11 @@ void checkContains(Type subType) {
    */
    abstract <T> T convert(List<GroupType> path, TypeConverter<T> converter);
 
+   public void setMetadata(Map<String, Object> metadata) {
```

Comment:
I would expect method comments describing the purpose and usage of this metadata. (We should mention that this metadata is for the current parquet-mr runtime only and it won't be serialized to the file.)

I understand this is the easiest way to add this map to the class but I don't really like it. For example by invoking this `setMetadata` the caller will not be informed if it overwrites any values already in. I would more like an approach where the user can set/get the metadata one-by-one.